File: 450-DSA-SHEET/ARRAYS/Python/8-Largest_Contiguous_Sum_Array.py
import math
from numpy import inf


class Solution:
    def maxsubarraysum(self,arr,N):
        curMax, maxTillNow = 0, -inf
        for i in arr:
            curMax = max(i, curMax+i)
            maxTillNow = max(maxTillNow, curMax)
        return maxTillNow

def main():
    T = int(input())
    while(T>0):
        n = int(input())
        arr = [int(x) for x in input().strip().split()]
        ob = Solution()
        print(ob.maxsubarraysum(arr,n))
        T-=1
    
if __name__ == "__main__":
    main()

File: 450-DSA-SHEET/ARRAYS/C-Plus-Plus/8-Largest_Contiguous_Sum_Array.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    long long maxsubarraysum(int arr[], int n){
        int curMax = arr[0], MaxTillNow = arr[0];
        for(int i=1;i<n;i++){
            curMax = max(arr[i], curMax+arr[i]);
            MaxTillNow = max(MaxTillNow, curMax);
        }
        return MaxTillNow;
    }
};

int main(){
    int t,n;
    cin>>t;
    while(t--){
        cin>>n;
        int arr[n];
        for(int i=0;i<n;i++){
            cin>>arr[i];
        }
        Solution ob;
        cout << ob.maxsubarraysum(arr,n) << endl;
    }
    return 0; 
}